operator/pkg/util/apiclient/idempotency.go

```diff
@@ -107,6 +107,8 @@ func CreateOrUpdateService(client clientset.Interface, service *corev1.Service)
 		}
 
 		service.ResourceVersion = older.ResourceVersion
+		service.Spec.ClusterIP = older.Spec.ClusterIP
+		service.Spec.ClusterIPs = older.Spec.ClusterIPs
```

Comment:
Looks good to me as we mostly use service with the `ClusterIP` type, and the IP can not be reset to none.
/assign @calvin0327